protocol/messenger_locker.go

```diff
@@ -0,0 +1,224 @@
+package protocol
+
+import (
+	"sync"
+
+	"github.com/status-im/status-go/protocol/encryption/multidevice"
+	"github.com/status-im/status-go/protocol/protobuf"
+)
+
+/*
+|--------------------------------------------------------------------------
+| chatMap
+|--------------------------------------------------------------------------
+|
+| A sync.Map wrapper for a specific mapping of map[string]*Chat
+|
+*/
+
+type chatMap struct {
+	sm sync.Map
+}
+
+func (cm *chatMap) Load(chatID string) (*Chat, bool) {
+	chat, ok := cm.sm.Load(chatID)
+	if chat == nil {
+		return nil, ok
+	}
+	return chat.(*Chat), ok
+}
+
+func (cm *chatMap) Store(chatID string, chat *Chat) {
+	cm.sm.Store(chatID, chat)
+}
+
+func (cm *chatMap) Range(f func(chatID string, chat *Chat) (shouldContinue bool)) {
+	nf := func(key, value interface{}) (shouldContinue bool) {
+		return f(key.(string), value.(*Chat))
+	}
+	cm.sm.Range(nf)
+}
+
+func (cm *chatMap) Delete(chatID string) {
+	cm.sm.Delete(chatID)
+}
+
+/*
+|--------------------------------------------------------------------------
+| contactMap
+|--------------------------------------------------------------------------
+|
+| A sync.Map wrapper for a specific mapping of map[string]*Contact
+|
+*/
+
+type contactMap struct {
+	sm sync.Map
+}
+
+func (cm *contactMap) Load(contactID string) (*Contact, bool) {
+	contact, ok := cm.sm.Load(contactID)
+	if contact == nil {
+		return nil, ok
+	}
+	return contact.(*Contact), ok
+}
+
+func (cm *contactMap) Store(contactID string, contact *Contact) {
+	cm.sm.Store(contactID, contact)
+}
+
+func (cm *contactMap) Range(f func(contactID string, contact *Contact) (shouldContinue bool)) {
+	nf := func(key, value interface{}) (shouldContinue bool) {
+		return f(key.(string), value.(*Contact))
+	}
+	cm.sm.Range(nf)
+}
+
+func (cm *contactMap) Delete(contactID string) {
+	cm.sm.Delete(contactID)
+}
+
+/*
+|--------------------------------------------------------------------------
```

Comment:
Haven't seen this style of commenting before. Any reason to introduce it?

Reply:
cc @Samyoul

Reply:
Generally when writing a file with a lot of repetitive structs I introduce these kinds of comments to form visual delimiter, these comments make reading through the file easier to see where the different sections of code are.

Reply:
I'm totally up for divider lines, maybe something like `/* --------- */` But think it would be convenient to put types description right before declarations to fit go documentation conventions. For example for later use with `godoc`

Reply:
Yeah, that's true